### ISSUE TITLE
debug and update examples

### ISSUE THREAD

**Issue by dbkeitel** [PyFstat/PyFstat#9]
@GregoryAshton had reported a specific failure in semi_coherent_search_using_MCMC.py back on 20190225 here: https://gitlab.aei.uni-hannover.de/GregAshton/PyFstat/issues/14

But I'll make this a general issue to check all examples.

**Comment by dbkeitel** [comment]
- The issue from https://gitlab.aei.uni-hannover.de/GregAshton/PyFstat/issues/14 doesn't show up for me any more.
- see #20 and #21 for new failures

**Comment by dbkeitel** [comment]
Oops, `examples/semicoherent_glitch_robust_directed_MCMC_search_on_1_glitch.py` got broken by #29:

```
18:12 INFO    : Exporting samples to example_data/glitch_robust_search/semicoherent_glitch_robust_directed_MCMC_search_on_1_glitch_samples.dat
Traceback (most recent call last):
  File "examples/glitch_examples/semicoherent_glitch_robust_directed_MCMC_search_on_1_glitch.py", line 80, in <module>
    mcmc.run()
  File "/home/dkeitel/git/sources/PyFstat/pyfstat/mcmc_based_searches.py", line 665, in run
    self.export_samples_to_disk()
  File "/home/dkeitel/git/sources/PyFstat/pyfstat/mcmc_based_searches.py", line 1591, in export_samples_to_disk
    " method.".format(Ncols, len(outfmt))
RuntimeError: Lengths of data rows (4) and output format (2) do not match. If your search class uses different keys than the base MCMCSearch class, override the get_savetxt_fmt method.
```

**Comment by dbkeitel** [comment]
`examples/grid_examples/grid_search_F0F1F2.py`:

```
18:20 INFO    : Saving data to example_data/grid_search_F0F1F2/grid_search_F0F1F2_NA_GridSearch.txt
Traceback (most recent call last):
  File "/home/dkeitel/bin/venvs/pyfstat/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py", line 146, in draw_path
    self._renderer.draw_path(gc, path, transform, rgbFace)
OverflowError: In draw_path: Exceeded cell block limit

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "examples/grid_examples/grid_search_F0F1F2.py", line 79, in <module>
    search.plot_1D(xkey="F2")
  File "/home/dkeitel/git/sources/PyFstat/pyfstat/grid_based_searches.py", line 341, in plot_1D
    fig.savefig(os.path.join(self.outdir, fname))
  File "/home/dkeitel/bin/venvs/pyfstat/lib/python3.7/site-packages/matplotlib/figure.py", line 2180, in savefig
    self.canvas.print_figure(fname, **kwargs)
  File "/home/dkeitel/bin/venvs/pyfstat/lib/python3.7/site-packages/matplotlib/backend_bases.py", line 2091, in print_figure
    **kwargs)
  File "/home/dkeitel/bin/venvs/pyfstat/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py", line 527, in print_png
    FigureCanvasAgg.draw(self)
  File "/home/dkeitel/bin/venvs/pyfstat/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py", line 388, in draw
    self.figure.draw(self.renderer)
  File "/home/dkeitel/bin/venvs/pyfstat/lib/python3.7/site-packages/matplotlib/artist.py", line 38, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/home/dkeitel/bin/venvs/pyfstat/lib/python3.7/site-packages/matplotlib/figure.py", line 1709, in draw
    renderer, self, artists, self.suppressComposite)
  File "/home/dkeitel/bin/venvs/pyfstat/lib/python3.7/site-packages/matplotlib/image.py", line 135, in _draw_list_compositing_images
    a.draw(renderer)
  File "/home/dkeitel/bin/venvs/pyfstat/lib/python3.7/site-packages/matplotlib/artist.py", line 38, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/home/dkeitel/bin/venvs/pyfstat/lib/python3.7/site-packages/matplotlib/axes/_base.py", line 2647, in draw
    mimage._draw_list_compositing_images(renderer, self, artists)
  File "/home/dkeitel/bin/venvs/pyfstat/lib/python3.7/site-packages/matplotlib/image.py", line 135, in _draw_list_compositing_images
    a.draw(renderer)
  File "/home/dkeitel/bin/venvs/pyfstat/lib/python3.7/site-packages/matplotlib/artist.py", line 38, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/home/dkeitel/bin/venvs/pyfstat/lib/python3.7/site-packages/matplotlib/lines.py", line 802, in draw
    renderer.draw_path(gc, tpath, affine.frozen())
  File "/home/dkeitel/bin/venvs/pyfstat/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py", line 148, in draw_path
    raise OverflowError("Exceeded cell block limit (set "
OverflowError: Exceeded cell block limit (set 'agg.path.chunksize' rcparam)
```

**Comment by dbkeitel** [comment]
`examples/transient_examples/long_transient_search_MCMC.py`:

```
18:52 INFO    : Exporting samples to example_data/long_transient/transient_search_samples.dat
Traceback (most recent call last):
  File "examples/transient_examples/long_transient_search_MCMC.py", line 55, in <module>
    mcmc.run()
  File "/home/dkeitel/git/sources/PyFstat/pyfstat/mcmc_based_searches.py", line 665, in run
    self.export_samples_to_disk()
  File "/home/dkeitel/git/sources/PyFstat/pyfstat/mcmc_based_searches.py", line 1591, in export_samples_to_disk
    " method.".format(Ncols, len(outfmt))
RuntimeError: Lengths of data rows (3) and output format (4) do not match. If your search class uses different keys than the base MCMCSearch class, override the get_savetxt_fmt method.
```

**Comment by dbkeitel** [comment]
- `examples/glitch_examples/semicoherent_glitch_robust_directed_MCMC_search_on_1_glitch.py` fixed by c8b3beb4c1ad983b1a7040b9a837918edea25de0
- `examples/grid_examples/grid_search_F0F1F2.py` fixed by 32d7976ebb0bf2aba4cece20e16c2f52c338c3bc
- `examples/transient_examples/long_transient_search_MCMC.py` fixed by 5dc4733bc4a2df296c986162b3c4b925b4514d23

That should be everything working for the moment.
